### ISSUE TITLE
Separate serving frontend assets via HTTP into separate package (fix error: cannot find module providing package github.com/simpleiot/simpleiot/assets/frontend)

### ISSUE THREAD

**Issue by bminer** [simpleiot/simpleiot#340]
It would be nice to import `simpleiot` without importing the generated frontend Elm assets. Separating these two components would be helpful, I think.

Somewhat related to #252

**Comment by cbrake** [comment]
We now get the following any time we `go get` simpleiot, which is a really bad developer experience:

```
github.com/simpleiot/simpleiot/assets/frontend: cannot find module providing package github.com/simpleiot/simpleiot/assets/frontend
```
Exploring various options of using go:embed, etc.

**Comment by cbrake** [comment]
took me a bit to figure out how to use it:

```
[cbrake@ceres ~]$ mkdir embed-test
[cbrake@ceres ~]$ cd embed-test
[cbrake@ceres embed-test]$ go mod init cliff
go: creating new go.mod: module cliff
[cbrake@ceres embed-test]$ go run git.icomputeconsulting.com/bminer/go-embed-test 
no required module provides package git.icomputeconsulting.com/bminer/go-embed-test; to add it:
        go get git.icomputeconsulting.com/bminer/go-embed-test
[cbrake@ceres embed-test]$ go get git.icomputeconsulting.com/bminer/go-embed-test
go: downloading git.icomputeconsulting.com/bminer/go-embed-test v0.0.0-20220418152551-6252db95000e
go: added git.icomputeconsulting.com/bminer/go-embed-test v0.0.0-20220418152551-6252db95000e
[cbrake@ceres embed-test]$ go run git.icomputeconsulting.com/bminer/go-embed-test/example
This is a test of //go:embed
```

Very neat -- thanks for the information!

And, nice git server BTW :-)

**Comment by cbrake** [comment]
so apparently you can't just run:

`go run git.icomputeconsulting.com/bminer/go-embed-test/example`

outside a go module directly.

But, if you specify the version, you can:

```
[cbrake@ceres ~]$ go run git.icomputeconsulting.com/bminer/go-embed-test/example@latest
This is a test of //go:embed
[cbrake@ceres ~]$ 
```

**Comment by cbrake** [comment]
This has been fixed in #438 -- we now check in the entire SIOT frontend into Git -- 52KB :-)
